sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -3392,4 +3397,180 @@ class HiveDDLSuite
       )
     }
   }
+
+  test("SPARK-52272: V2SessionCatalog does not alter schema on Hive Catalog") {
+    val externalCatalog = new CustomHiveCatalog(spark.sessionState.catalog.externalCatalog)
```

Comment:
why do we need a `CustomHiveCatalog` to test it? I think we can get the real v2 session catalog via `session.sessionState.catalogManager.v2SessionCatalog` and test it.